core/src/main/java/com/microsoft/applicationinsights/internal/quickpulse/QuickPulse.java

```diff
@@ -149,13 +149,14 @@ public synchronized void stop(long timeout, TimeUnit timeUnit) {
         try {
             thread.join();
         } catch (InterruptedException e) {
-            InternalLogger.INSTANCE.trace("Stack trace generated is %s", ExceptionUtils.getStackTrace(e));
+            Thread.currentThread().interrupt();
```

Comment:
Maybe better to `interrupt()` both threads first, then try to `join()` them, and if either throws `InterruptedException`, then re-`interrupt()` and `return` from method (without setting `initialized` to `false` at the bottom?)

Reply:
I think `initialized` should still be set to false after interrupting the worker threads. Both workers have already had signals to stop their runnables.